pkg/proxy/iptables/proxier.go

```diff
@@ -123,20 +123,28 @@ type serviceInfo struct {
 	serviceLBChainName       utiliptables.Chain
 }
 
+//FIXME:
 // returns a new proxy.ServicePort which abstracts a serviceInfo
-func newServiceInfo(port *v1.ServicePort, service *v1.Service, baseInfo *proxy.BaseServiceInfo) proxy.ServicePort {
-	info := &serviceInfo{BaseServiceInfo: baseInfo}
-
-	// Store the following for performance reasons.
-	svcName := types.NamespacedName{Namespace: service.Namespace, Name: service.Name}
-	svcPortName := proxy.ServicePortName{NamespacedName: svcName, Port: port.Name}
-	protocol := strings.ToLower(string(info.Protocol()))
-	info.serviceNameString = svcPortName.String()
-	info.servicePortChainName = servicePortChainName(info.serviceNameString, protocol)
-	info.serviceFirewallChainName = serviceFirewallChainName(info.serviceNameString, protocol)
-	info.serviceLBChainName = serviceLBChainName(info.serviceNameString, protocol)
-
-	return info
+func newServiceInfo(hasher hasher) func(port *v1.ServicePort, service *v1.Service, baseInfo *proxy.BaseServiceInfo) proxy.ServicePort {
+	return func(port *v1.ServicePort, service *v1.Service, baseInfo *proxy.BaseServiceInfo) proxy.ServicePort {
+		info := &serviceInfo{BaseServiceInfo: baseInfo}
+
+		// Store the following for performance reasons.
+		svcName := types.NamespacedName{Namespace: service.Namespace, Name: service.Name}
+		svcPortName := proxy.ServicePortName{NamespacedName: svcName, Port: port.Name}
+		protocol := strings.ToLower(string(info.Protocol()))
+		info.serviceNameString = svcPortName.String()
+		info.servicePortChainName = servicePortChainName(hasher, info.serviceNameString, protocol)
+		info.serviceFirewallChainName = serviceFirewallChainName(hasher, info.serviceNameString, protocol)
+		info.serviceLBChainName = serviceLBChainName(hasher, info.serviceNameString, protocol)
+
+		return info
+	}
+}
+
+//FIXME:
+type hasher interface {
+	Hash(input ...string) string
```

Comment:
Get rid of the interface? `type Hasher func(...string) string` ? It would simplify both the real and test implementations